### ISSUE TITLE
Crash when deleting all messages in threaded view

### ISSUE THREAD

**Issue by ervee** [thunderbird/thunderbird-android#1494]
### Expected behavior

K9-mail should not crash when deleting all messages in a thread.
### Actual behavior

K9-mail crashes when deleting all messages in a thread. If you delete all but one and then the last one then it goes fine.
### Steps to reproduce
1. Enable threaded view
2. Enter a thread with a bunch (10 or so) of messages
3. Select all messages and delete them
### Environment

K-9 Mail version:5.010

Android version:
6.0.1

Account type (IMAP, POP3, WebDAV/Exchange):
IMAP (dovecot)
### Logcat

```
07-08 21:38:56.122 10519 10519 E AndroidRuntime: java.lang.IllegalStateException: attempt to re-open an already-closed object: SQLiteQuery: SELECT m.id AS id,uid,internal_date,subject,date,sender_list,to_list,cc_list,read,flagged,answered,forwarded,attachment_count,folder_id,preview,root,name FROM threads t JOIN messages m ON (m.id = t.message_id) LEFT JOIN folders f ON (m.folder_id = f.id) WHERE root = ? AND deleted = 0 AND (empty IS NULL OR empty != 1) ORDER BY read ASC, date DESC, m.id DESC
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:55)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:58)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:151)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:123)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:236)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:197)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:197)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:197)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at com.fsck.k9.cache.EmailProviderCacheCursor.moveToPosition(EmailProviderCacheCursor.java:112)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:197)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.CursorAdapter.getItemId(CursorAdapter.java:258)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.HeaderViewListAdapter.getItemId(HeaderViewListAdapter.java:194)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.AbsListView$RecycleBin.retrieveFromScrap(AbsListView.java:6868)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.AbsListView$RecycleBin.getScrapView(AbsListView.java:6598)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.AbsListView.obtainView(AbsListView.java:2345)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.ListView.makeAndAddView(ListView.java:1876)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.ListView.fillDown(ListView.java:702)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.ListView.fillSpecific(ListView.java:1367)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.ListView.layoutChildren(ListView.java:1663)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.AbsListView.onLayout(AbsListView.java:2148)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at com.android.internal.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:493)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2685)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.View.layout(View.java:16722)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewGroup.layout(ViewGroup.java:5438)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2175)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1935)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1111)
07-08 21:38:56.122 10519 10519 E AndroidRuntime:    at android.view.ViewRootImpl$TraversalRunnable.
```

**Comment by cketti** [comment]
Note to developers:
I don't think there's an easy fix for this bug. But I want to get rid of the `CursorLoader` anyway. That should fix things.

**Comment by cketti** [comment]
This should be fixed in recent beta versions (5.7xx).
